instrumentation/apache-httpclient/apache-httpclient-4.3/library/src/main/java/io/opentelemetry/instrumentation/apachehttpclient/v4_3/HostAndRequestAsHttpUriRequest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.apachehttpclient.v4_3;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpRequest;
+import org.apache.http.ProtocolVersion;
+import org.apache.http.RequestLine;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.message.AbstractHttpMessage;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/** Wraps HttpHost and HttpRequest into a HttpUriRequest for tracers and injectors. */
+final class HostAndRequestAsHttpUriRequest extends AbstractHttpMessage implements HttpUriRequest {
```

Comment:
I need to adopt @trask's new pattern for this class but otherwise think this PR is good for an initial look